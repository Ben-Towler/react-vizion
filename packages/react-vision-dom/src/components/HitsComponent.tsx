import React from 'react';
import DefaultHitComponent from './DefaultHitComponent';

export type SearchResult = {
  resourceId: string;
  address: any;
  distance: number;
  geoPoint: {
    lat: number;
    lng: number;
  };
  resourceName: string;
  note: string;
  onlineBookingUrl: string;
  phones: any;
  type: string;
};

type Props = {
  results: SearchResult[];
  className?: string;
  style?: React.CSSProperties;
  noResultsFound: React.ReactNode;
  hit: (result: SearchResult) => React.ReactNode;
};

const HitsComponent: React.FunctionComponent<Props> = ({
  results,
  className,
  style,
  hit,
  noResultsFound,
}) => {
  return Array.isArray(results) && results.length > 0 ? (
    <div className={className} style={style}>
      {results.map(searchResult => {
        hit ? (
          hit(searchResult)
        ) : (
          <DefaultHitComponent searchResult={searchResult} />
        );
      })}
    </div>
  ) : (
    <>
      {noResultsFound || <div className="hit-no-results-found-component"></div>}
    </>
  );
};

export default HitsComponent;
