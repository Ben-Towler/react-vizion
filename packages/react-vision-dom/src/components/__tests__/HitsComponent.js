import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import HitsComponent from '../HitsComponent';

Enzyme.configure({ adapter: new Adapter() });

describe('HitsComponent', () => {
  it('applies its default props', () => {
    const instance = renderer.create(<HitsComponent refine={() => null} />);

    expect(instance.toJSON()).toMatchSnapshot();

    instance.unmount();
  });

  it('should render with default hit component', () => {});

  it('should render with default no results found', () => {});

  describe('Style tests', () => {
    it('should render main div with className props ', () => {});
    it('should render main div with style props ', () => {});
  });

  describe('Custom inner components', () => {
    it('should render with custom hit component', () => {});

    it('should render with custom noResultsFound component', () => {});
  });
});
