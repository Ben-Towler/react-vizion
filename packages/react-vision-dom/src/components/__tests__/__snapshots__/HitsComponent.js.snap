// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HitsComponent applies its default props 1`] = `
<div
  className="hit-no-results-found-component"
/>
`;
